graphics/opengl/system_headers.h

```diff
@@ -24,35 +24,69 @@
 
 #include "common/scummsys.h"
 
-#ifdef USE_GLES2
+// On OS X we only support GL contexts. The reason is that Apple's GL interface
+// uses "void *" for GLhandleARB which is not type compatible with GLint. This
+// kills our aliasing trick for extension functions and thus would force us to
+// supply two different Shader class implementations or introduce other
+// wrappers. OS X only supports GL contexts right now anyway (at least
+// according to SDL2 sources), thus it is not much of an issue.
+#if defined(MACOSX) && (!defined(USE_GLES_MODE) || USE_GLES_MODE != 0)
+//#warning "Only forced OpenGL mode is supported on Mac OS X. Overriding settings."
+#undef USE_GLES_MODE
+#define USE_GLES_MODE 0
+#endif
 
-#define GL_GLEXT_PROTOTYPES
-#ifdef IPHONE
-#include <OpenGLES/ES2/gl.h>
-#include <OpenGLES/ES2/glext.h>
+// We allow to force GL or GLES modes on compile time.
+// For this the USE_GLES_MODE define is used. The following values represent
+// the given selection choices:
+//  0 - Force OpenGL context
+//  1 - Force OpenGL ES context
+//  2 - Force OpenGL ES 2.0 context
+#ifdef USE_GLES_MODE
+	#define USE_FORCED_GL    (USE_GLES_MODE == 0)
+	#define USE_FORCED_GLES  (USE_GLES_MODE == 1)
+	#define USE_FORCED_GLES2 (USE_GLES_MODE == 2)
 #else
-#include <GLES2/gl2.h>
-#include <GLES2/gl2ext.h>
+	#define USE_FORCED_GL    0
+	#define USE_FORCED_GLES  0
+	#define USE_FORCED_GLES2 0
 #endif
-#undef GL_GLEXT_PROTOTYPES
 
-#ifndef GL_BGRA
-#	define GL_BGRA GL_BGRA_EXT
-#endif
+#if USE_FORCED_GLES2
 
-#if !defined(GL_UNPACK_ROW_LENGTH)
-// The Android SDK does not declare GL_UNPACK_ROW_LENGTH_EXT
-#define GL_UNPACK_ROW_LENGTH 0x0CF2
-#endif
+	#define GL_GLEXT_PROTOTYPES
+	#if defined(IPHONE)
+		#include <OpenGLES/ES2/gl.h>
+		#include <OpenGLES/ES2/glext.h>
+	#elif defined(__ANDROID__)
+		#include <GLES/gl.h>
```

Comment:
we should not include gles1 for gles2. any reason to do it?

Reply:
Indeed, it was a bad cut paste from backends/graphics/opengl/opengl-sys.h where GLES is included for Android.
During the merge of the two files I didn't took attention about this.
Now my question will be: should we try to support platforms using GLESv1 only and without GLAD?
I don't think we need to and from what I understand, it seems we don't support it today.

Reply:
gles1 support could be explored in future. dunno if we ever going to support gles1 platforms only, they might have cpu to slow. about GLAD, it could be supported depending on OS. anyway we could ignore it for now.